src/main/java/com/school/project/service/impl/SubjectServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.school.project.service.impl;
+
+import com.school.project.exception.SubjectNotFoundException;
+import com.school.project.model.entities.Subject;
+import com.school.project.repository.SubjectRepository;
+import com.school.project.service.SubjectService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@Service
+public class SubjectServiceImpl implements SubjectService {
+
+    @Autowired
+    SubjectRepository subjectRepository;
+
+    @Override
+    public Subject createSubject(Subject subject) {
+        subjectRepository.save(subject);
```

Comment:
Please return here what you will get from repository and not what you are getting from parameters